readthedocs/doc_builder/backends/sphinx.py

```diff
@@ -168,14 +154,10 @@ def get_config_params(self):
 
     def append_conf(self, **__):
         """
-        Find or create a ``conf.py`` and appends default content.
+        Find a ``conf.py`` and appends default content.
 
         The default content is rendered from ``doc_builder/conf.py.tmpl``.
         """
-        if self.config_file is None:
-            master_doc = self.create_index(extension='rst')
-            self._write_config(master_doc=master_doc)
```

Comment:
We should raise an exception at this point https://github.com/rtfd/readthedocs.org/blob/43c1ba7e8b65bc3bb5a8b2d7b7f6de94dfa91b44/readthedocs/projects/models.py#L725-L725

Reply:
@stsewd Do you mean we should keep this
https://github.com/rtfd/readthedocs.org/blob/30583646e470b1bdcf606c52e851835d6495df20/readthedocs/doc_builder/backends/sphinx.py#L176
then raise the error or just raise the error if `self.config_file is None`

Reply:
I feel that create index should stay, that's part of https://github.com/rtfd/readthedocs.org/issues/1800, we can address that in another PR

Reply:
(with or without config file)